pkg/restapi/vc/operation/operations_test.go

```diff
@@ -126,15 +142,203 @@ func TestCreateCredentialHandler(t *testing.T) {
+
+	urlVars := make(map[string]string)
+	urlVars[profilePathVariable] = notFoundID
+
+	req = mux.SetURLVars(req, urlVars)
+
+	t.Run("get profile success", func(t *testing.T) {
+		profile := createProfileSuccess(t, op)
+
+		r, err := http.NewRequest(http.MethodGet,
+			"/profile/"+profile.ID,
+			bytes.NewBuffer([]byte("")))
+		require.NoError(t, err)
+
+		rr := httptest.NewRecorder()
+
+		urlVars := make(map[string]string)
+		urlVars[profilePathVariable] = profile.ID
+		req = mux.SetURLVars(r, urlVars)
+
+		getProfileHandler.Handle().ServeHTTP(rr, req)
+
+		require.Equal(t, http.StatusOK, rr.Code)
+		profileResponse := &ProfileResponse{}
+		err = json.Unmarshal(rr.Body.Bytes(), profileResponse)
+		require.NoError(t, err)
+		require.Equal(t, profileResponse.ID, profile.ID)
+		require.Equal(t, profileResponse.URL, profile.URL)
+	})
+	t.Run("get profile error, bad request", func(t *testing.T) {
+		req, err := http.NewRequest(http.MethodGet,
+			"/profile/"+notFoundID,
+			bytes.NewBuffer([]byte("")))
+		require.NoError(t, err)
+		rr := httptest.NewRecorder()
+		getProfileHandler.Handle().ServeHTTP(rr, req)
+		require.Equal(t, http.StatusBadRequest, rr.Code)
+	})
+	t.Run("get profile error, bad request", func(t *testing.T) {
+		req, err := http.NewRequest(http.MethodGet,
+			"/profile/"+notFoundID,
+			bytes.NewBuffer([]byte("")))
+		require.NoError(t, err)
+		rw := mockResponseWriter{}
+		getProfileHandler.Handle().ServeHTTP(rw, req)
+		require.Contains(t, logContents.String(),
+			"Failed to write response for profile retrieval failure: response writer failed")
+	})
+}
+
+func createProfileSuccess(t *testing.T, op *Operation) *ProfileResponse {
+	req, err := http.NewRequest(http.MethodPost, createProfileEndpoint, bytes.NewBuffer([]byte(testIssuerProfile)))
+	require.NoError(t, err)
+
+	rr := httptest.NewRecorder()
+
+	createProfileEndpoint := getHandler(t, op, createProfileEndpoint)
+	createProfileEndpoint.Handle().ServeHTTP(rr, req)
+
+	profile := &ProfileResponse{}
+
+	err = json.Unmarshal(rr.Body.Bytes(), &profile)
+	require.NoError(t, err)
+
+	require.Equal(t, http.StatusCreated, rr.Code)
+	require.NotEmpty(t, profile.ID)
+
+	return profile
+}
 func TestOperation_WriteResponse(t *testing.T) {
-	svc := New()
-	require.NotNil(t, svc)
-	svc.writeResponse(&httptest.ResponseRecorder{}, "token")
+	op, err := New(memstore.NewProvider())
+	require.NoError(t, err)
+	require.NotNil(t, op)
+	op.writeResponse(&httptest.ResponseRecorder{}, "token")
+}
+
+func TestOperation_PrepareURL(t *testing.T) {
+	resp, err := prepareURL("&&%^)$")
+	require.Empty(t, resp)
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "failed to parse the uri")
 }
 
 func TestCreate(t *testing.T) {
 	b := mockResponseWriter{}
-	op := New()
+	op, err := New(memstore.NewProvider())
+	require.NoError(t, err)
 	req, err := http.NewRequest(http.MethodPost, createCredentialEndpoint,
 		bytes.NewBuffer([]byte(testCreateCredentialRequest)))
 	require.NoError(t, err)
```

Comment:
1. b isn't used
2. This test looks like it's only testing functions from external libraries (memstore.NewProvider() and http.NewRequest), so this test isn't needed

Reply:
It is used 	op.createCredentialHandler(b, req) and it increases the coverage by 1.5 percent.

Reply:
Oops, the rest of the code was collapsed (I only saw until line 317).